lib/streamlit/connections/sql_connection.py

```diff
@@ -80,7 +82,9 @@ def _connect(self, autocommit: bool = False, **kwargs) -> "Engine":
 
         kwargs = deepcopy(kwargs)
         conn_param_kwargs = extract_from_dict(_ALL_CONNECTION_PARAMS, kwargs)
-        conn_params = ChainMap(conn_param_kwargs, self._secrets.to_dict())
+        conn_params = (
```

Comment:
@sfc-gh-jcarroll heads up that this change wasn't mentioned in the spec, but I'm assuming we'll want consistent behavior across connections so will want to do this here too

Reply:
Hmmm... so the main issue I wanted to avoid was unexpected implicit merging of config across contexts - it came up when debugging connection errors and seeing it was unexpected for streamlit secrets and snowflake config in different places to be joined together, and more difficult to debug.

I'm not sure you have the same problem with kwargs - it's pretty intuitive (to me at least?) that kwargs will be merged to config and its a normal pattern for other connection libraries (AWS etc). WDYT? Happy to discuss more and get some wider feedback too, I'm not strongly entrenched here.

Reply:
Ah, yeah that makes sense. I think if this is a standard pattern across connection libraries, then we should probably keep things as-is. The change is a one-liner anyway, so reviving it is trivial if we decide that we do want to avoid merging the two